src/sql/base/browser/ui/table/plugins/checkboxSelectColumn.plugin.ts

```diff
@@ -259,38 +256,52 @@ export class CheckboxSelectColumn<T extends Slick.SlickData> implements Slick.Pl
 			return this.checkboxTemplateCustom(row);
 		}
 
-		return this._selectedRowsLookup[row]
-			? strings.format(checkboxTemplate, 'checked', this.getAriaLabel(true))
-			: strings.format(checkboxTemplate, '', this.getAriaLabel(false));
+		// If checkbox is a row selector, we don't have requirement to enable/disable it, so always leave it enabled
+		return this.getCheckboxHtml(this._selectedRowsLookup[row], this._options.title, true);
 	}
 
 	checkboxTemplateCustom(row: number): string {
+		const propertyValue = this.getCheckboxPropertyValue(row);
 		// use state after toggles
 		if (this._useState) {
-			return this._selectedCheckBoxLookup[row]
-				? strings.format(checkboxTemplate, 'checked', this.getAriaLabel(true))
-				: strings.format(checkboxTemplate, '', this.getAriaLabel(false));
+			return this.getCheckboxHtml(this._selectedCheckBoxLookup[row], this._options.title, propertyValue.enabled);
 		}
 
 		// use data for first time rendering
 		// note: make sure Init is called before using this._grid
-		let rowVal = this._grid?.getDataItem(row);
-		if (rowVal && this._options.title && rowVal[this._options.title] === true) {
+		if (propertyValue.checked) {
 			this._selectedCheckBoxLookup[row] = true;
-			return strings.format(checkboxTemplate, 'checked', this.getAriaLabel(true));
 		}
 		else {
 			delete this._selectedCheckBoxLookup[row];
-			return strings.format(checkboxTemplate, '', this.getAriaLabel(false));
 		}
+		return this.getCheckboxHtml(propertyValue.checked, this._options.title, propertyValue.enabled);
 	}
 
 	private isCustomActionRequested(): boolean {
 		return (this._options.actionOnCheck === ActionOnCheck.customAction);
 	}
 
-	private getAriaLabel(checked: boolean): string {
-		return checked ? `"${this._options.title} ${nls.localize("tableCheckboxCell.Checked", "checkbox checked")}"` :
```

Comment:
this is not necessary, the screen reader will read the type (checkbox) and the checked state, we shouldn't need to provide it as part of the aria label.